torch_geometric/explain/algorithm/pgm_explainer.py

```diff
@@ -0,0 +1,462 @@
+import logging
+from typing import List, Optional, Tuple
+
+import numpy as np
+import torch
+
+from torch_geometric.explain.config import ModelTaskLevel
+from torch_geometric.explain.explanation import Explanation
+from torch_geometric.utils import k_hop_subgraph
+from torch_geometric.utils.subgraph import get_num_hops
+
+from .base import ExplainerAlgorithm
+
+
+class PGMExplainer(ExplainerAlgorithm):
+    r"""The PGMExplainer model from the `"PGMExplainer: Probabilistic
+    Graphical Model Explanations  for Graph Neural Networks"
+    <https://arxiv.org/abs/1903.03894>`_ paper
+
+    Args:
+        perturb_feature_list (List): indices of the perturbed features
+            for graph classification explanations
```

Comment:
Isn't this used for both graph and node tasks?

Reply:
The authors of the paper only use this for their graph explanations (I think? it should work for node tasks as well -- can add if that makes more sense)

Reply:
It looks like you already support that.  `explain_graph` uses `perturbh_feature_on_nodes` which uses `feature_index` (`perturb_feature_list` as its called now).